tests/test_baseInterface.py

```
@@ -0,0 +1,604 @@
+        """
+        m1 = DummyManager()
+        m1.a, m1.b = 1, 2
+        m2 = DummyManager()
+        m2.a, m2.b = 1, 3
+        bucket = DummyBucket(self.manager_class, [m1, m2])
+        grp = bucket.group_by("a", "b")
+        self.assertIsInstance(grp, GroupBucket)
+        self.assertEqual(grp._manager_class, self.manager_class)
+
+        self.assertEqual(grp._manager_class, self.manager_class)
+        self.assertEqual(getattr(grp, "_group_by_keys"), ("a", "b"))
```

Ruff (0.11.9)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)